integrationtests/src/test/java/io/kroxylicious/proxy/ProxyRpcTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+
+import io.kroxylicious.test.client.KafkaClient;
+import io.kroxylicious.test.server.MockServer;
+
+import static io.kroxylicious.proxy.Utils.startProxy;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+public class ProxyRpcTest {
+
+    public static final int PROXY_PORT = 9192;
+    public static final String PROXY_ADDRESS = "localhost:" + PROXY_PORT;
+    private final ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+    @Test
+    public void test() throws Exception {
+        MockTestDefinition testDefinition = readYaml("rpc_scenarios.yml");
+        try (MockServer mockServer = MockServer.startOnRandomPort(); KafkaClient kafkaClient = new KafkaClient()) {
+            String config = configure(testDefinition, mockServer);
+            try (KafkaProxy ignored = startProxy(config)) {
+                for (MockTestDefinition.Scenario scenario : testDefinition.getScenarios()) {
+                    testScenario(mockServer, kafkaClient, scenario);
+                }
+            }
+        }
+    }
+
+    private static void testScenario(MockServer mockServer, KafkaClient kafkaClient, MockTestDefinition.Scenario scenario) {
+        try {
+            mockServer.setResponse(scenario.givenMockResponse());
+            CompletableFuture<JsonNode> future = kafkaClient.get("localhost", PROXY_PORT, scenario.whenSendRequest());
+            JsonNode response = future.get(10, TimeUnit.SECONDS);
+            assertEquals(scenario.thenMockReceivesRequest(), onlyRequest(mockServer), "unexpected request received at mock for scenario: " + scenario.name());
```

Comment:
I imagine the expectations could be better expressed as a `diff` like we've done with multitenant, but didn't want to keep adding stuff to this PR.

So in the YAML you could express:
1. the mock response
2. the request to send to kroxy
3. the difference between the request to kroxy and the request received at mock
4. the difference between the mock response and the response received by the client

Reply:
> I've implemented tests for four RPCs, generating the test data is a bit manual so I'll hold off doing a test for every RPC until this approach gets accepted or rejected

Okay I've changed direction here a little, @showuon mentioned in passing we could generate the RPCs instead of manually supplying them in a file (which would be a manual thing to deal as new ApiMessages are added). So now instead of being driven by a big YAML file the initial test is driven by randomly generated request/response examples for every ApiMessage type. So it covers all ApiMessage types except a handful that aren't proxied/intercepted the same way as the majority (SASL requests, ApiVersions).

Reply:
Made redundant by moving away from YAML for testing